### DISCUSSION TITLE
syncPermissions() doesn't work in my application

### DISCUSSION THREAD

**Discussion by KaikSilva** [spatie/laravel-permission#1822] [Q&A]
I'm making an application in laravel .... where I use syncPermissions() to insert the conditions, but syncPermissions() doesn't do the insert into the database...

the code :

    public function permissionsSync(Request $request, $role)
    {
    $permissionsRequest = $request->except(['_token', '_method']); //here I separate what I will use
   
        foreach($permissionsRequest as $key => $value) {
        $permissions[] = Permission::where('id', $key)->first(); //here I get the requested permissions from the database
        }
   
        $role = Role::where('id', $role)->first(); //Here I get the Role where the permissions will be assigned

        if(!empty($permissions)){ //here valid if it is empty

            $role->syncPermissions($permissions); //It would be Here where I do the insert

        } else {

            $role->syncPermissions(null);

        }
   
    return redirect()->route('admin.role.permissions', ['role' => $role->id]); //I redirect the screen
    }

I've already done the debug and all variables are filled as expected... Thanks!

**Comment by erikn69** [comment]
Any exception?

Why? `$permissions[] = Permission::where('id', $key)->first();` 
Why send unused value on post  `$value`(`$key => $value`)

Better 
```php
$permissions =Permission::whereIn('id',  array_keys($permissionsRequest))->get();
```
Even better just 
```php
$permissions = array_keys($permissionsRequest);
```
